Device/UIDeviceExtension.swift

```diff
@@ -212,32 +206,40 @@ public enum DeviceType: String, EnumProtocol {
     }
 }
 
+// MARK: -
 
-// MARK: - EnumProtocol
+public extension UIDevice {
 
-internal protocol EnumProtocol: Hashable {
-    /// -returns: All Enum Values
-    static var all: [Self] { get }
+	/// The `DeviceType` of the device in use
+	public var deviceType: DeviceType {
+		return DeviceType.current
+	}
 }
 
+#if swift(>=4.2)
```

Comment:
Will it compile [this](https://github.com/schickling/Device.swift/pull/39/files#diff-50c20e5593c02bb42bba54d6d05f69ffR17) line for Swift upper version (upper than 4.2)?